lib/common/strings.c

```diff
@@ -599,24 +599,42 @@ crm_str_table_dup(GHashTable *old_table)
 
 /*!
  * \internal
- * \brief Add a word to a space-separated string list
+ * \brief Add a word to a string list of words
  *
- * \param[in,out] list  Pointer to beginning of list
- * \param[in]     word  Word to add to list
+ * \param[in,out] list       Pointer to current string list (may not be NULL)
+ * \param[in,out] len        If not NULL, must be set to length of \p list,
+ *                           and will be updated to new length of \p list
+ * \param[in]     word       String to add to \p list
+ * \param[in]     separator  String to separate words in \p list
  *
- * \return (Potentially new) beginning of list
- * \note This dynamically reallocates list as needed.
+ * \note This dynamically reallocates \p list as needed. \p word may contain
+ *       \p separator, though that would be a bad idea if the string needs to be
+ *       parsed later.
  */
-char *
-pcmk__add_word(char *list, const char *word)
+void
+pcmk__add_separated_word(char **list, size_t *len, const char *word,
+                         const char *separator)
```

Comment:
What happens if separator is NULL?

Reply:
Whoops :) What do you think: should it default to "" or " "?

Reply:
A space is probably what we'd want most of the time.  I doubt it'll ever come up given the wrapper, but still.